crates/rpc-types-eth/src/fee.rs

```diff
@@ -45,6 +45,7 @@ pub struct FeeHistory {
     pub base_fee_per_gas: Vec<u128>,
     /// An array of block gas used ratios. These are calculated as the ratio
     /// of `gasUsed` and `gasLimit`.
+    #[cfg_attr(feature = "serde", serde(deserialize_with = "gas_used_ratio_deser::deserialize"))]
```

Comment:
I believe we have this in alloy-serde:

https://github.com/alloy-rs/alloy/blob/caa380d188da17d3795b7d4f540609c1b64ff3c5/crates/serde/src/optional.rs#L7-L7